imap_processing/lo/l1b/lo_l1b.py

```diff
@@ -483,6 +491,66 @@ def set_coincidence_type(
     return l1b_de
 
 
+def convert_tofs_to_eu(
+    l1a_de: xr.Dataset,
+    l1b_de: xr.Dataset,
+    attr_mgr_l1a: ImapCdfAttributes,
+    attr_mgr_l1b: ImapCdfAttributes,
+) -> xr.Dataset:
+    """
+    Convert the TOFs to engineering units.
+
+    The TOFs are converted from data numbers (DN) to engineering units (EU) using the
+    following equation:
+    TOF_EU = C0 + C1 * TOF_DN
+
+    where C0 and C1 are the conversion coefficients for each TOF.
+
+    This equation is applied to all four TOFs (TOF0, TOF1, TOF2, TOF3).
+
+    Parameters
+    ----------
+    l1a_de : xarray.Dataset
+        The L1A DE dataset.
+    l1b_de : xarray.Dataset
+        The L1B DE dataset.
+    attr_mgr_l1a : ImapCdfAttributes
+        Attribute manager used to get the fill values for the L1A DE dataset.
+    attr_mgr_l1b : ImapCdfAttributes
+        Attribute manager used to get the fill values for the L1B DE dataset.
+
+    Returns
+    -------
+    l1b_de : xarray.Dataset
+        The L1B DE dataset with the TOFs converted to engineering units.
+    """
+    tof_fields = ["tof0", "tof1", "tof2", "tof3"]
+    tof_conversions = [TOF0_CONV, TOF1_CONV, TOF2_CONV, TOF3_CONV]
+
+    # Loop through the TOF fields and convert them to engineering units
+    for tof, conv in zip(tof_fields, tof_conversions):
+        # Get the fill value for the L1A and L1B TOF
+        fillval_1a = attr_mgr_l1a.get_variable_attributes(tof)["FILLVAL"]
+        fillval_1b = attr_mgr_l1b.get_variable_attributes(tof)["FILLVAL"]
+        # Create a mask for the TOF
+        mask = l1a_de[tof] != fillval_1a
+        # convert the DE TOF to engineering units
+        tof_eu = np.where(
+            mask,
+            conv.C0 + conv.C1 * l1a_de[tof],
+            fillval_1b,
+        )
+        # Add the EU TOF to the dataset. If the TOF is not present, set it to the fill
+        # value for the L1B TOF data.
```

Comment:
The second sentence should be above the where statement and merged into your other comment up there I believe since you are doing that work within where, not the dataset creation.